### ISSUE TITLE
Sharpness of images at varying zoom

### ISSUE THREAD

**Issue by thomasaarholt** [ArturKovacs/emulsion#9]
Thanks for making this! Really nice.

In comparison with the standard Windows 10, Emulsion looks a bit blurred on the following screenshot. See the scale bar at the bottom left in particular. I'm sure there are several strategies for smoothing and anti-aliasing, what does Emulsion use? [Original png attached](https://user-images.githubusercontent.com/2721423/79915963-34f12a80-8428-11ea-8b84-6c19f4ec1ce8.png).

![image](https://user-images.githubusercontent.com/2721423/79912799-e5f4c680-8422-11ea-8ba0-cd782b6f672b.png)

**Comment by ArturKovacs** [comment]
Indeed. This is something that I don't have experience with, although this is definitely within the scope of my interests so all I'm saying is that there may be someone else who could tackle this more efficiently than I could.

**Comment by ArturKovacs** [comment]
The question in the post somehow eluded my attention. Emulsion currently generates up to four mip-maps and then it uses "LinearMipmapLinear" filtering for minification and "Linear" filtering for magnification by default. "Nearest" is used for maginfication if the renderend texel size is greater than  some threshold.

**Comment by Aloso** [comment]
I noticed this bug as well:

![Screenshot_20200506_194037](https://user-images.githubusercontent.com/15658558/81210820-bcb36900-8fd2-11ea-9227-56a829022641.png)

Images are, when scaled down, slightly more blurry than they should be. This is most obvious when the image contains text:

![Help screen blurry](https://user-images.githubusercontent.com/15658558/81211621-ee78ff80-8fd3-11ea-884c-6bc9ecdd6123.png)

The right image is emulsion showing the help dialog. The left image is the same image, opened in emulsion normally.

I wonder if the method used to render the help dialog could be used for other downscaled images as well.

**Comment by ArturKovacs** [comment]
The info screen uses regular linear filtering for minification. Nevertheless I can get the two to be exactly as sharp, but it's super sensitive to the scale. Just one pixel off and the one displayed as an image is significantly blurrier.

I made some quick tests to see if this can be improved by simple switching the filtering mode. And indeed it improved by a lot when using NearestMipmapLinear. The photos app still seems to do better but we are pretty close now.

From left-to-right: Old Emulsion, Win Photos, New Emulsion

![image](https://user-images.githubusercontent.com/8320264/81215609-e328d280-8fd9-11ea-97d9-d83444f07211.png)

**Comment by Aloso** [comment]
I tried this, and makes the image sharper, but breaks antialiasing:

![nearest_mipmap_linear](https://user-images.githubusercontent.com/15658558/81216792-cb524e00-8fdb-11ea-8bc9-14e0057d9bb7.png)

**Comment by Aloso** [comment]
It's much better now. Only below a certain zoom threshold, it looks as before.

**Comment by Shnatsel** [comment]
[Krita](https://krita.org/) spent a lot of time on a good interpolation method for their OpenGL canvas mode. Perhaps the approach or even the code can be copied if the license allows.
